### PR TITLE
Issue943/pull deps neo4j

### PR DESCRIPTION
**Pull request by Leahh02**
- [x] beeflow core pull-deps should use a Dockerfile to build the needed neo4j container and add the apoc file
- [x] This dockerfile should be placed in beeflow/data/dockerfiles and should be used by the ci integration tests so we only need to modify it in one place.
- [x] Also, if a user uses "beeflow core pull-deps" to have their own dependent containers, they should be used in the bee.conf file. Consider automatically editting it and removing the ~/.beeflow/deps directory. Beeflow should not be running when this is done, so adding a check for that should probably be done.

### CODE REVIEW

**Comment by Leahh02** [comment]
I'm working on the last part right now. I can change the name of beeflow/data/dockerfiles/Dockerfile.apoc_neo4j if needed. I can also remove beeflow/data/dockerfiles/Dockerfile.neo4j-ppc64le if needed.

**Comment by pagrubel** [comment]
> I'm working on the last part right now. I can change the name of beeflow/data/dockerfiles/Dockerfile.apoc_neo4j if needed. I can also remove beeflow/data/dockerfiles/Dockerfile.neo4j-ppc64le if needed.

Yes, I think Dockerfile.neo4j would suffice and yes please remove the other. We will need to probably add one for arm architecture in the future.

File: beeflow/client/core.py [hunk 1]
```diff
@@ -35,6 +37,9 @@
 from beeflow.common.deps import redis_manager
 
 
+REPO_PATH = Path(*Path(__file__).parts[:-3])
```
Comment (pagrubel) [inline beeflow/client/core.py:40]: Will this work for beeflow when published in PyPi?

**Review by pagrubel — APPROVED** [review]
This is a good refactor of utilities and tests used while incorporating pulling dependent containers used for BEE. It is a nice feature to update the config file to use the newly pulled dependent containers. My only concern is how the path to the Dockerfile will work once published in pip. In order to test that I'm going to merge this and publish it in a pre-release.

**Comment by rstyd** [comment]
Looks good to me!

**Comment by pagrubel** [comment]
Just a note. If someone has already started beeflow with other dependent containers, it will continue to use them until they delete the bee_workdir/deps or perform a beeflow core reset, but this can be another issue.

**Pull request merged**